src/utils/general.ts

```diff
@@ -122,76 +125,94 @@ function evaluate(
 
 const aggregateMap: {
   [key in QueryViewerAggregationType]: (
-    values: number[],
+    values: GxBigNumber[],
     quantities: number[]
-  ) => number;
+  ) => GxBigNumber;
 } = {
   [QueryViewerAggregationType.Sum]: (
-    values: number[],
+    values: GxBigNumber[],
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     _quantities: number[]
   ) => {
-    let sumValues: number = null;
+    let sumValues: GxBigNumber = null;
 
     for (let i = 0; i < values.length; i++) {
       if (values[i]) {
-        sumValues += values[i];
+        sumValues = sumValues ? add(sumValues, values[i]) : values[i];
       }
     }
+
     return sumValues;
   },
 
   [QueryViewerAggregationType.Average]: (
-    values: number[],
+    values: GxBigNumber[],
     quantities: number[]
   ) => {
-    let sumValues: number = null;
+    let sumValues: GxBigNumber = null;
     let sumQuantities: number = null;
 
     for (let i = 0; i < values.length; i++) {
-      if (values[i]) {
-        sumValues += values[i];
+      const value = values[i];
+      if (value) {
+        sumValues = sumValues ? add(sumValues, value) : value;
         sumQuantities += quantities[i];
       }
     }
-    return sumValues != null ? sumValues / sumQuantities : null;
+
+    return sumValues != null ? divide(sumValues, sumQuantities) : null;
   },
 
   [QueryViewerAggregationType.Count]: (
-    _values: number[],
+    _values: GxBigNumber[],
     quantities: number[]
-  ) => quantities.reduce((a, b) => a + b, 0),
+  ) => {
+    return new GxBigNumber(quantities.reduce((a, b) => a + b, 0));
+  },
```

Comment:
```suggestion
  ) => 
    new GxBigNumber(quantities.reduce((a, b) => a + b, 0));
```

Reply:
Hi Nico, thanks for the review!, I was actually tryng to uploated a created branch, so this way be able to collaborate in the feature, but I couldn't do it. Maybe I need some permissions?

Reply:
I've sent you an invitation.